crates/ruff_linter/resources/test/fixtures/ruff/RUF066.py

```diff
@@ -0,0 +1,50 @@
+from typing import Annotated, Optional, TypeAlias, Union
+
+# Errors
+a: Union[Annotated[int, "An integer"], None] = 5
+b: Annotated[Union[int, str], "An integer or string"] | None = "World"
+c: Optional[Annotated[float, "A float"]] = 2.71
+d: Union[Annotated[int, "An integer"], Annotated[str, "A string"]]
+
+def f1(x: Union[Annotated[int, "An integer"], None]) -> None:
+    pass
+
+def f2(y: Annotated[Union[int, str], "An integer or string"] | None) -> None:
+    pass
+
+def f3(z: Optional[Annotated[float, "A float"]]) -> None:
+    pass
+
+e: Annotated[Annotated[int, "An integer"], "Another annotation"]
```

Comment:
This being an error implies that `list[Annotated[...]]` or similar would also produce an error. That part seems fine to me, but I wonder if it would be worth having a separate error message for these cases since the mention of union/optional doesn't make sense here? Would also like to have test cases showing that behavior one way or the other.